core/src/array/array_sorted_read_state.cc

```diff
@@ -2437,30 +2088,57 @@ Status ArraySortedReadState::read_dense_sorted_col() {
     return array_->read_default(
         copy_state_.buffers_, copy_state_.buffer_sizes_);
 
-  // Iterate over each tile slab
+  // Handle resume copy
+  if (resume_copy_) {
```

Comment:
Yeah I was thinking about something similar. I did this with `advance_cell_slab`. I need to make the methods `static` to pass them as arguments.